src/Dofs/DofHandler.jl

```diff
@@ -766,45 +766,72 @@ getfieldinterpolation(fh::FieldHandler, field_idx::Int) = fh.field_interpolation
 getfieldinterpolation(fh::FieldHandler, field_name::Symbol) = getfieldinterpolation(fh, find_field(fh, field_name))
 
 """
-    reshape_to_nodes(dh::AbstractDofHandler, u::Vector{T}, fieldname::Symbol) where T
+    evaluate_at_grid_nodes(dh::AbstractDofHandler, u::Vector{T}, fieldname::Symbol) where T
 
-Reshape the entries of the dof-vector `u` which correspond to the field `fieldname` in nodal order.
-Return a matrix with a column for every node and a row for every dimension of the field.
-For superparametric fields only the entries corresponding to nodes of the grid will be returned. Do not use this function for subparametric approximations.
-"""
-function reshape_to_nodes(dh::DofHandler, u::Vector{T}, fieldname::Symbol) where T
-    # make sure the field exists
-    fieldname ∈ getfieldnames(dh) || error("Field $fieldname not found.")
+Evaluate the approximated solution for field `fieldname` at the node
+coordinates of the grid given the Dof handler `dh` and the solution vector `u`.
 
-    field_dim = getfielddim(dh, fieldname)
-    space_dim = field_dim == 2 ? 3 : field_dim
-    data = fill(T(NaN), space_dim, getnnodes(dh.grid))  # set default value
+Return a vector of length `getnnodes(grid)` where entry `i` contains the evalutation of the
+approximation in the coordinate of node `i`. If the field does not live on parts of the
+grid, the corresponding values for those nodes will be returned as `NaN`s.
+"""
+function evaluate_at_grid_nodes(dh::DofHandler, u::Vector, fieldname::Symbol)
+    return _evaluate_at_grid_nodes(dh, u, fieldname)
+end
 
+# Internal method that have the vtk option to allocate the output differently
+function _evaluate_at_grid_nodes(dh::DofHandler, u::Vector{T}, fieldname::Symbol, ::Val{vtk}=Val(false)) where {T, vtk}
+    # Make sure the field exists
+    fieldname ∈ getfieldnames(dh) || error("Field $fieldname not found.")
+    # Figure out the return type (scalar or vector)
+    field_idx = find_field(dh, fieldname)
+    ip = getfieldinterpolation(dh, field_idx)
+    if vtk
```

Comment:
Would it be good to call this `pad_zero` or something like that?

Reply:
Maybe, it doesn't just enable padeing, also makes the output a matrix, for example.

Reply:
Yes, probably not the best naming suggestion either; As it is internal I guess it can be changed later if there are cases for which it makes sense (I was just a bit on the side that it would be nice not to have references to the vtk-format at different places in the code-base when a part-goal of #692 is to centralize that to one place)

I suppose it is too inefficient to first create the "standard" output and then reformat it for vtk?